### PR TITLE
Misc code cleanup

### PR DESCRIPTION
**Pull request by hlinnaka**
See commit messages for details.

### CODE REVIEW

File: src/backend/utils/sort/tuplestorenew.c [hunk 1]
```diff
@@ -1234,39 +1234,6 @@ bool ntuplestore_acc_current_tupleslot(NTupleStoreAccessor *tsa, TupleTableSlot
 	return true;
 }
 
-bool ntuplestore_acc_current_data(NTupleStoreAccessor *tsa, void **data, int *len)
-{
-	bool fOK = ntuplestore_acc_current_data_internal(tsa, (void **) data, len);
```
Comment (danielgustafsson) [inline src/backend/utils/sort/tuplestorenew.c:1239]: After this, there is only one caller of `ntuplestore_acc_current_data_internal` left and IMO reading the code, it makes more sense to inline `ntuplestore_acc_current_data_internal` into `ntuplestore_acc_current_tupleslot`.

File: src/backend/cdb/cdbsreh.c [hunk 2]
```diff
@@ -324,18 +324,6 @@ sendnumrows_internal(int64 numrejected, int64 numcompleted)
...
- *
- * Using this function the QE sends back to the client QD the number
- * of rows that were rejected in this last data load in SREH mode.
- */
-void
-SendNumRowsRejected(int64 numrejected)
-{
-	sendnumrows_internal(numrejected, 0);
-}
-
```
[the remark this answers is not in the document]
Comment (hlinnaka) [reply src/backend/cdb/cdbsreh.c:338]: Makes sense, fixed.

File: src/backend/utils/init/test/postinit_test.c [hunk 3]
```diff
@@ -9,14 +9,6 @@
...
 #define PG_RE_THROW() siglongjmp(*PG_exception_stack, 1)
 
-#define errfinish errfinish_impl
-
-static int
-errfinish_impl(int dummy pg_attribute_unused(),...)
-{
-	PG_RE_THROW();
-}
-
```
[the remark this answers is not in the document]
Comment (hlinnaka) [reply src/backend/utils/init/test/postinit_test.c]: Yes. And the PR pipeline proved you right, the unittest was now failing. Reverted that change.

File: src/backend/utils/sort/tuplestorenew.c (hunk 1 above)
Comment (hlinnaka) [reply src/backend/utils/sort/tuplestorenew.c:1239]: Yeah, I think you're right. However, I'm actually just working on removing tuplestorenew.c altogether, replacing all uses of it with the regular PostgreSQL tuplestore.c. tuplestorenew.c has the extra feature that the tuplestore can be shared across processes, but that's pretty straightforward to add to tuplestore.c, too. So right now, I can't be bothered to do anything in tuplestorenew.c, except for removing stuff.

Comment (danielgustafsson) [reply src/backend/utils/sort/tuplestorenew.c:1239]: Right, that makes sense, I retract my comment.

**Pull request merged**